### DISCUSSION TITLE
请教有关日志模块的问题

### DISCUSSION THREAD

**Discussion by guang19** [chenshuo/muduo#559] [General]
我看源码发现muduo的log设计是一条日志一条stream，且log buffer是在栈空间分配的，用完立刻就会释放掉，速度很快。我想问的是如果前台有大量的log请求，即使栈空间设置的很大，是否仍有可能造成栈溢出？且较为频繁的分配和释放内存是否也会出现内存碎片？如果改为堆分配，除了性能差些，是否会存在其他问题？

**Comment by chenshuo** [comment]
如果你有以上担心：
* 你能不能试着写一个程序，通过“前台有大量的log请求”，造成“栈溢出"的后果？
* 你能不能试着写一个程序，通过在栈空间分配释放内存，造成“内存碎片”？

**Comment by guang19** [comment]
好的，谢谢大神指教
